File: app/src/main/java/com/example/david/lists/widget/MyRemoteViewsService.java
package com.example.david.lists.widget;

import android.content.Intent;
import android.widget.RemoteViewsService;

import com.example.david.lists.widget.di.DaggerMyRemoteViewsServiceComponent;

public class MyRemoteViewsService extends RemoteViewsService {
    @Override
    public RemoteViewsFactory onGetViewFactory(Intent intent) {
        return DaggerMyRemoteViewsServiceComponent.builder()
                .application(getApplication())
                .intent(intent)
                .build()
                .remoteViewsFactory();
    }
}

File: app/src/main/java/com/precopia/david/lists/widget/view/MyRemoteViewsService.kt
package com.precopia.david.lists.widget.view

import android.content.Intent
import android.widget.RemoteViewsService

import com.precopia.david.lists.widget.view.buildlogic.DaggerMyRemoteViewsServiceComponent

class MyRemoteViewsService : RemoteViewsService() {
    override fun onGetViewFactory(intent: Intent) =
            DaggerMyRemoteViewsServiceComponent.builder()
                    .application(application)
                    .intent(intent)
                    .build()
                    .remoteViewsFactory()
}
